fedbiomed/common/db.py

```diff
@@ -0,0 +1,51 @@
+# This file is originally part of Fed-BioMed
+# SPDX-License-Identifier: Apache-2.0
+
+'''
+Interfaces with the node component database.
+'''
+
+from tinydb.table import Table, Document
+
+
+def cast_(func):
+    """Decorator function for typing casting"""
+    # Wraps TinyDb get, all, search and upsert methods
+    def wrapped(*args, **kwargs):
+        add_docs = kwargs.get("add_docs")
+        if add_docs is not None:
+            kwargs.pop("add_docs")
+
+        document = func(*args, **kwargs)
+        if isinstance(document, list):
+            casted = [dict(r) for r in document]
+        elif isinstance(document, Document):
+            casted = dict(document)
+        else:
+            # Plain python type 
+            return document
```

Comment:
we should always return 2 values when `add_doc` is `True` eg:

```
else:
    casted = document
```

pushed a fix for that